File: mobile/lib/domain/entities/activity_entity.dart
/**
 * @file activity_entity.dart
 * @description 活動領域模型 / Activity domain entity
 * @description_zh 定義行程每一站活動的核心業務資料結構
 * @description_en Defines the core business data structure of an activity in a trip
 */

class ActivityEntity {
  final String id;
  final String tripId;
  final String time;
  final String title;
  final String? subtitle;
  final String? content;
  final String type;
  final String? iconName;
  final Map<String, dynamic>? personalInfo;
  final int sortOrder;
  final String? locationName;
  final double? latitude;
  final double? longitude;
  final List<String>? imageUrls;

  ActivityEntity({
    required this.id,
    required this.tripId,
    required this.time,
    required this.title,
    this.subtitle,
    this.content,
    required this.type,
    this.iconName,
    this.personalInfo,
    this.sortOrder = 0,
    this.locationName,
    this.latitude,
    this.longitude,
    this.imageUrls,
  });
}

File: backend/src/modules/trips/entities/activity.entity.ts
import { Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, UpdateDateColumn, ManyToOne, JoinColumn } from 'typeorm';
import { Trip } from './trip.entity';

/**
 * @file activity.entity.ts
 * @description 活動實體 / Activity Entity
 * @description_zh 定義行程內每一項活動的詳細表結構
 * @description_en Defines the table structure for every activity within a trip
 */

@Entity('activities')
export class Activity {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ type: 'uuid' })
  trip_id: string;

  @ManyToOne(() => Trip, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'trip_id' })
  trip: Trip;

  @Column({ length: 20 })
  time: string;

  @Column({ length: 100 })
  title: string;

  @Column({ length: 255, nullable: true })
  subtitle: string;

  @Column({ type: 'text', nullable: true })
  content: string;

  @Column({ length: 30 })
  type: string;

  @Column({ length: 50, nullable: true })
  icon_name: string;

  @Column({ type: 'jsonb', nullable: true })
  personal_info: any;

  @Column({ default: 0 })
  sort_order: number;

  @Column({ length: 255, nullable: true })
  location_name: string;

  @Column({ type: 'decimal', precision: 10, scale: 8, nullable: true })
  latitude: number;

  @Column({ type: 'decimal', precision: 11, scale: 8, nullable: true })
  longitude: number;

  @Column({ type: 'text', nullable: true })
  image_url: string;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;
}
